src/ape/utils/trace.py

```diff
@@ -26,6 +40,22 @@
 _DEFAULT_INDENT = 2
 
 
+class TreeLike(Protocol):
+    address: Any
+    calls: Iterable
+    calldata: Any
+    returndata: Any
+    failed: bool
+    call_type: Any
+    gas_cost: int
+    value: int
+    gas_limit: int
```

Comment:
I don't think `Protocol` will work quite right here

This might need a revisit...

Reply:
Can you explain more?

Reply:
I think this is getting registered as class variables, not fields

Reply:
It does not need to be fields because it is not a base class of anything.

Look at this example:

```python
from typing import Protocol

class Stocky(Protocol):
    stock_level: int

class Corn:
    stock_level = 1
    something_else = 99

class Wheat:
    stock_level = 2
    something_else_else = 99

def grind(stalk: Stocky):
    print(stalk.stock_level)


if __name__ == "__main__":
    corn_0 = Corn()
    corn_1 = Corn()
    corn_1.stock_level = 3
    grind(corn_0)
    grind(corn_1)
    grind(Wheat())
```

it outputs:

```
1
3
2
```

Notice that the classes `Corn` and `Wheat` are **not** subclasses of the `Stocky` protocol - they merely adhere to it. And the method `grind()` takes a value that adheres to that protocol.

If I run `mypy` on the script, there are no issues.